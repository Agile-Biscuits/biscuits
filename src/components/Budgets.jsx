import styled from '@emotion/styled';
import Budget from './Budget';

const Container = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  gap: 16px;
`;

export default function Budgets({ budgets }) {
  return (
    <Container>
      {budgets.sort((a, b) => a.value / a.amount - b.value / b.amount).map((
        { name, amount, value },
      ) => <Budget name={name} amount={amount} value={value} key={name} />)}
    </Container>
  );
}
