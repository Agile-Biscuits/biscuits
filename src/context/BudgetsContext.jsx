import { createContext, useState, useEffect } from 'react';
import budgets from '../../fixtures/budgets';

export const BudgetsContext = createContext();

export const BudgetsProvider = ({ children }) => {
  const [budgetsData, setBudgetsData] = useState([]);

  useEffect(() => {
    setBudgetsData(budgets); // set your budgets data when the component mounts
  }, []);

  return (
    <BudgetsContext.Provider value={{ budgets: budgetsData, setBudgets: setBudgetsData }}>
      {children}
    </BudgetsContext.Provider>
  );
};
