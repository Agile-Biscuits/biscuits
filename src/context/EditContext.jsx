import { createContext, useState } from 'react';

export const EditContext = createContext({
  isEditing: false,
  setIsEditing: () => {}
});

export const EditProvider = ({ children }) => {
  const [isEditing, setIsEditing] = useState(false);

  return (
    <EditContext.Provider value={{ isEditing, setIsEditing }}>
      {children}
    </EditContext.Provider>
  );
};
