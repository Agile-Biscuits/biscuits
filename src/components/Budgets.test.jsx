import { render, cleanup } from '@testing-library/react';
import Budgets from './Budgets';
import { EditProvider } from '../context/EditContext';
import { BrowserRouter } from 'react-router-dom';

afterEach(cleanup);

describe('Budgets', () => {
  it('should render the budgets', () => {
    const budgets = [
      { name: 'Test 1', amount: 100, value: 50 },
      { name: 'Test 2', amount: 200, value: 50 },
    ];
    expect(
      render(
        <BrowserRouter>
          <EditProvider>
            <Budgets budgets={budgets} />
          </EditProvider>
        </BrowserRouter>,
      ),
    ).toBeTruthy();
  });
  it('renders correct number of budgets', () => {
    const budgets = [
      { name: 'Test 1', amount: 100, value: 50 },
      { name: 'Test 2', amount: 200, value: 50 },
    ];
    const { getAllByTestId } = render(
      <BrowserRouter>
        <EditProvider>
          <Budgets budgets={budgets} />
        </EditProvider>
      </BrowserRouter>,
    );
    expect(getAllByTestId('budget').length).toBe(budgets.length);
  });
  it('matches snapshot', () => {
    const budgets = [
      { name: 'Test 1', amount: 100, value: 50 },
      { name: 'Test 2', amount: 200, value: 50 },
    ];
    const { asFragment } = render(
      <BrowserRouter>
        <EditProvider>
          <Budgets budgets={budgets} />
        </EditProvider>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
