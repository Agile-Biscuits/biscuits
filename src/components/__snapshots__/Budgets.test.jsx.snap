// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Budgets matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1zblq7-Container eo0izpr0"
  >
    <a
      class="css-1srxvew-StyledLink e1dz7f3g0"
      href="/transactions/Test 2"
      style="text-decoration: none;"
    >
      <div
        class="css-2hu90h-Container e1dz7f3g8"
        data-testid="budget"
      >
        <div
          class="css-cwb82f-Fill e1dz7f3g1"
          value="50"
        />
        <div
          class="css-15xwpwu-InfoContainer e1dz7f3g7"
        >
          <div
            class="css-njdknn-NameContainer e1dz7f3g3"
          >
            <div
              class="css-kh8k75-Name e1dz7f3g6"
            >
              TEST 2
            </div>
          </div>
          <div
            class="css-7nkki1-Value e1dz7f3g2"
          >
            £50.00
          </div>
        </div>
      </div>
    </a>
    <a
      class="css-1srxvew-StyledLink e1dz7f3g0"
      href="/transactions/Test 1"
      style="text-decoration: none;"
    >
      <div
        class="css-2hu90h-Container e1dz7f3g8"
        data-testid="budget"
      >
        <div
          class="css-1qtj0e0-Fill e1dz7f3g1"
          value="50"
        />
        <div
          class="css-15xwpwu-InfoContainer e1dz7f3g7"
        >
          <div
            class="css-njdknn-NameContainer e1dz7f3g3"
          >
            <div
              class="css-kh8k75-Name e1dz7f3g6"
            >
              TEST 1
            </div>
          </div>
          <div
            class="css-7nkki1-Value e1dz7f3g2"
          >
            £50.00
          </div>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
